sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -353,10 +385,10 @@ void joint_matrix_copy(
     Group sg, joint_matrix<Group, T1, Use1, Rows, Cols, Layout1> &src,
     joint_matrix<Group, T2, Use2, Rows, Cols, Layout2> &dst) {
 #if defined(__SYCL_DEVICE_ONLY__)
-#if defined(__NVPTX__)
+#if defined(__NVPTX__) || defined(__HIP_PLATFORM_AMD_MFMA__)
   std::ignore = sg;
-  for (int i = 0; i < src.cuda_impl.wi_marray.size(); i++) {
-    dst.cuda_impl.wi_marray[i] = src.cuda_impl.wi_marray[i];
+  for (int i = 0; i < src.matrix_impl.wi_marray.size(); i++) {
```

Comment:
```suggestion
dst.matrix_impl.wi_marray = src.matrix_impl.wi_marray;
```
Is equivalent and simpler.

Reply:
thanks @JackAKirk , done.